app/src/main/java/com/example/tkey_android/FirstFragment.java

```diff
@@ -163,557 +222,642 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
 
-            builder.setPositiveButton("OK", (dialog, which) -> {
-                String password = passwordEditText.getText().toString();
-                // Handle the entered password
-                showLoading();
-                try {
-                    String question = "what's your password?";
-                    SecurityQuestionModule.changeSecurityQuestionAndAnswer(activity.tKey, question, password, result -> {
-                        if (result instanceof Result.Error) {
-                            requireActivity().runOnUiThread(() -> {
-                                renderError(((Result.Error<Boolean>) result).exception);
-                                hideLoading();
-                            });
-                        } else if (result instanceof Result.Success) {
-                            requireActivity().runOnUiThread(() -> {
-                                try {
-                                    Boolean changed = ((Result.Success<Boolean>) result).data;
-                                    if (changed) {
-                                        String setAnswer = SecurityQuestionModule.getAnswer(activity.tKey);
-                                        binding.changePassword.setEnabled(false);
-                                        Snackbar snackbar = Snackbar.make(view1, "Password changed to" + setAnswer, Snackbar.LENGTH_LONG);
-                                        snackbar.show();
-                                        hideLoading();
-                                    } else {
-                                        Snackbar snackbar = Snackbar.make(view1, "Password failed to be changed", Snackbar.LENGTH_LONG);
-                                        snackbar.show();
-                                        hideLoading();
-                                    }
-                                } catch (RuntimeError e) {
-                                    Snackbar snackbar = Snackbar.make(view1, "A problem occurred: " + e, Snackbar.LENGTH_LONG);
-                                    snackbar.show();
-                                    hideLoading();
-                                }
-                            });
-                        }
-                    });
-                } catch (Exception e) {
-                    renderError(e);
-                    hideLoading();
-                }
-            });
+                                    tssNonce = TSSModule.getTSSNonce(activity.tKey, tag, false);
 
-            builder.setNegativeButton("Cancel", (dialog, which) -> dialog.cancel());
-            AlertDialog dialog = builder.create();
-            dialog.show();
-        });
+                                    TSSModule.getTSSShare(activity.tKey, tag, factorKey, 0, _result -> {
+                                        if (_result instanceof Result.Error) {
+                                            System.out.println("Could not create tagged tss shares for tkey");
```

Comment:
These errors should be displayed or thrown and not just printed to the console, else the user will never see them.